web/src/main/java/org/springframework/security/web/csrf/CookieCsrfTokenRepository.java

```diff
@@ -47,6 +49,14 @@
 
 	private String cookieName = DEFAULT_CSRF_COOKIE_NAME;
 
+	private Method setHttpOnlyMethod;
+
+	private boolean cookieHttpOnly = true;
+
+	public CookieCsrfTokenRepository() {
+		this.setHttpOnlyMethod = ReflectionUtils.findMethod(Cookie.class, "setHttpOnly", boolean.class);
```

Comment:
Yes, you are correct. HttpOnly attribute was introduced in Servlet 3.0. Using this reflective pattern allows us to support older versions of the servlet spec.